functions/v2/http-logging/main.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# [START functions_structured_logging]
+import google.cloud.logging
+from google.cloud.logging.handlers import StructuredLogHandler
+from google.cloud.logging_v2.handlers import setup_logging
+
+
+def structured_logging(request):
+    logging_client = google.cloud.logging.Client()
+
+    handler = StructuredLogHandler()
+    setup_logging(handler)
+
+    logger = logging_client.logger("New-Structured-Log")
+
+    msg = (f"{request.headers}"
+           f"{request.get_data().decode()}"
+           )
+
+    # Write structured log with HTTP request data
+    logger.log_struct(
```

Comment:
to print to stdout, I believe we need to import the standard logging library. Will check with the Observability DPEs

Reply:
Yeah, that's correct. The `google.cloud.logging.logger` methods always send logs over the network. `log_struct` in this case means "log JSON payload".

It's the standard logging library integration that supports standard out logs. Since you already ran `setup_logs()`, you should now be able to do: 

```
import logging
logging.info("Hello World")
```

Reply:
I'm not entirely sure what you want to show for the `component` field, but you can either create a [JSON payload](https://github.com/googleapis/python-logging/blob/8a67b73cdfcb9da545671be6cf59c724360b1544/UPGRADING.md#full-json-log-support-in-standard-library-integration-316-339-447) like this:

```
data_dict = {"component": "arbitrary-property"}
logging.info("Hello World", extra={"json_fields": data_dict})
```

or this:

```
data_dict = {"message": "Hello World", component": "arbitrary-property"}
logging.info(json.dumps(data_dict))
```

Or you can attach it as a [label](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry), which is probably the better way to add extra user-defined key-value metadata that isn't exactly part of the core payload

```
labels =  {"component": "arbitrary-property"}
logging.info("Hello World", extra={"labels": labels})
```